keras_nlp/layers/transformer_decoder_test.py

```diff
@@ -29,32 +29,40 @@ def test_valid_call(self):
             intermediate_dim=4,
             num_heads=2,
         )
-        output = decoder(encoder_input, decoder_input)
+        output = decoder(decoder_input, encoder_input)
+        # should raise ValueError if encoder_input is not provided
+        try:
```

Comment:
Remove the try block. You can add a separate test for these using `self.assertRaises(ValueError)`. There's other examples in this test file.